specreduce/line_matching.py

```diff
@@ -0,0 +1,121 @@
+from typing import Sequence
+
+import numpy as np
+
+import astropy.units as u
+from astropy.stats import gaussian_fwhm_to_sigma, gaussian_sigma_to_fwhm
+from astropy.modeling import models
+from astropy.table import QTable
+from astropy.wcs import WCS as astropy_WCS
+
+from gwcs.wcs import WCS as gWCS
+
+from specutils import Spectrum1D
+from specutils.fitting import find_lines_threshold, fit_lines
+
+
+__all__ = [
+    "find_arc_lines",
+    "match_lines_wcs"
+]
+
+
+def find_arc_lines(
+    spectrum: Spectrum1D,
+    fwhm: float | u.Quantity = 5.0 * u.pix,
+    window: float = 3.0,
+    noise_factor: float = 5.0
+) -> QTable:
+    """
+    Find arc lines in a spectrum using `~specutils.fitting.find_lines_threshold` and
+    then perform gaussian fits to each detected line to refine position and FWHM.
+
+    Parameters
+    ----------
+    spectrum : The extracted arc spectrum to search for lines. It should be background-subtracted
+        and must have an "uncertainty" attribute.
+
+    fwhm : Estimated full-width half-maximum of the lines in pixels.
+
+    window : The window size in units of fwhm to use for the gaussian fits.
+
+    noise_factor : The factor to multiply the uncertainty by to determine the noise threshold
+        in the `~specutils.fitting.find_lines_threshold` routine.
+
+    Returns
+    -------
+    QTable
+        A table of detected arc lines and their properties.
```

Comment:
Describe what table columns are returned (seems like just pixel_center and wavelength?)